pkg/logproto/compat.go

```diff
@@ -214,3 +214,91 @@ func init() {
 	jsoniter.RegisterTypeEncoderFunc("logproto.LegacySample", SampleJsoniterEncode, func(unsafe.Pointer) bool { return false })
 	jsoniter.RegisterTypeDecoderFunc("logproto.LegacySample", SampleJsoniterDecode)
 }
+
+// Combine unique values from multiple LabelResponses into a single LabelResponse.
+// TODO(jordanrushing): Should we sort labels lexicographically before returning?
+func MergeLabelResponses(responses []*LabelResponse) (*LabelResponse, error) {
+	if len(responses) == 0 {
+		return nil, fmt.Errorf("no label responses to merge")
```

Comment:
is this an error? what about just an empty labels response? `&LabelsResponse{Values []string{}}`?

Reply:
Good point.